### PR TITLE
Show user profiles in the user page header

### PR DESCRIPTION
**Pull request by Half-Shot**
This is to alleviate an issue where bridges will show ugly looking userIds in order to ensure uniqueness, and so the UI should prefer to highlight the displayname / profile where possible.

I don't know if we were planning to use the room name / room avatar as the true source instead of the profile of the user, so feel free to ask and I can change that.

### CODE REVIEW

File: src/Client.js [hunk 1]
```diff
@@ -595,6 +608,17 @@ class Client {
...
 
+    thumbnailLink(mxcUri, method, width, height) {
+        if (!mxcUri) {
+            return;
+        }
+        if (mxcUri.indexOf("mxc://") !== 0) {
+            return;
+        }
+        const mediaUrl = this.serverUrl.slice(0, -1 * "/client".length);
+        return `${mediaUrl}/media/r0/thumbnail/${mxcUri.split("mxc://")[1]}?method=${method}&width=${width}&height=${height}`;
```
Comment (kegsay) [inline src/Client.js]: please `encodeURIComponent` the method/width/height.

File: src/UserPage.js [hunk 2]
```diff
@@ -50,6 +51,17 @@ class UserPage extends React.Component {
         let roomId;
         try {
             roomId = await this.props.client.followUser(this.props.userId);
+            try {
+                const userProfile = await this.props.client.getProfile(this.props.userId);
+                this.setState({
+                    userProfile,
```
Comment (kegsay) [inline src/UserPage.js:57]: This is valid JS?

**Review by kegsay — APPROVED** [review]
Otherwise LGTM. I need to write some script to automatically build/run this then screenshot the UI..

File: src/UserPage.js (hunk 2 above)
Comment (Half-Shot) [reply src/UserPage.js:57]: https://alligator.io/js/object-property-shorthand-es6/ yup :)

**Pull request merged**